src/Data/Validation.hs

```diff
@@ -203,16 +202,18 @@ instance (NFData e, NFData a) => NFData (Validation e a) where
       Failure e -> rnf e
       Success a -> rnf a
 
--- | 'validate's the @a@ with the given predicate, returning @e@ if the predicate does not hold.
+-- | 'validate's an @a@ producing an updated optional value, returning
+-- @e@ in the empty case.
 --
 -- This can be thought of as having the less general type:
 --
 -- @
--- validate :: e -> (a -> Bool) -> a -> Validation e a
+-- validate :: e -> (a -> Maybe b) -> a -> Validation e b
 -- @
-validate :: Validate v => e -> (a -> Bool) -> a -> v e a
-validate e p a =
-  if p a then _Success # a else _Failure # e
+validate :: Validate v => e -> (a -> Maybe b) -> a -> v e b
+validate e p a = case p a of
```

Comment:
Why not `maybe (_Failure # b) ((#) _Success) . p`? Just out of interest, I'm only a tourist in this repo :blush:

Reply:
I did think about this variation first, but I decided against it doesn't bring much to the code. It doesn't pass my personal golfing threshold of "is this more elegant than a pattern match?", and it also wasn't idiomatic when compared to the rest of the code.

Reply:
That's fair! I'd go for a birdie and write `maybe (_Failure # e) (\b -> _Success # b) . p` but this version is also readable and elegant :100:

Reply:
a birdie 😆